### PR TITLE
[new release] anthill (0.1)

### PR DESCRIPTION
**Pull request by martindemello**
Word search library and utility

- Project page: <a href="https://github.com/martindemello/anthill">https://github.com/martindemello/anthill</a>
- Documentation: <a href="https://martindemello.github.io/anthill/">https://martindemello.github.io/anthill/</a>

##### CHANGES:

* Initial release

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
There is apparently a missing dependency (I forgot to include it in my fix):
```
#=== ERROR while compiling anthill.0.1 ========================================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.10.0 | pinned(https://github.com/martindemello/anthill/releases/download/0.1/anthill-0.1.tbz)
# path                 ~/.opam/4.10/.opam-switch/build/anthill.0.1
# command              ~/.opam/4.10/bin/dune build -p anthill -j 72
# exit-code            1
# env-file             ~/.opam/log/anthill-23-618dce.env
# output-file          ~/.opam/log/anthill-23-618dce.out
### output ###
# File "bin/dune", line 5, characters 18-25:
# 5 |  (preprocess (pps lwt_ppx)))
#                       ^^^^^^^
# Error: Library "lwt_ppx" not found.
# Hint: try: dune external-lib-deps --missing -p anthill -j 72 @install
# File "lib/dune", line 6, characters 18-25:
# 6 |  (preprocess (pps lwt_ppx)))
#                       ^^^^^^^
# Error: Library "lwt_ppx" not found.
# Hint: try: dune external-lib-deps --missing -p anthill -j 72 @install
```

**Comment by kit-ty-kate** [comment]
That should be fine now. Let's wait for the CI to finish. Could you return the fix upstream?

**Comment by kit-ty-kate** [comment]
There seems to be yet another issue on OCaml 4.06. Could you have a look at it?
```
#=== ERROR while compiling anthill.0.1 ========================================#
# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(https://github.com/martindemello/anthill/releases/download/0.1/anthill-0.1.tbz)
# path                 ~/.opam/4.06/.opam-switch/build/anthill.0.1
# command              ~/.opam/4.06/bin/dune build -p anthill -j 72
# exit-code            1
# env-file             ~/.opam/log/anthill-23-d51e6e.env
# output-file          ~/.opam/log/anthill-23-d51e6e.out
### output ###
# File "/home/opam/.opam/4.06/lib/ppx_inline_test/ppx_inline_test.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
# File "/home/opam/.opam/4.06/lib/ppx_expect/ppx_expect.dune", line 1, characters 0-0:
# Warning: .dune files are ignored since 2.0. Reinstall the library with dune
# >= 2.0 to get rid of this warning and enable support for the subsystem this
# library provides.
#       ocamlc lib/.anthill.objs/byte/anthill__Group.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.06/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.anthill.objs/byte -I /home/opam/.opam/4.06/lib/base -I /home/opam/.opam/4.06/lib/base/caml -I /home/opam/.opam/4.06/lib/base/md5 -I /home/opam/.opam/4.06/lib/base/shadow_stdlib -I /home/opam/.opam/4.06/lib/bin_prot -I /home/opam/.opam/4.06/lib/bin_prot/shape -I /home/opam/.opam/4.06/lib/bytes -I /home/opam/.opam/4.06/lib/camomile -I /home/opam/.opam/4.06/lib/camomile/default_config -I /home/opam/.opam/4.06/lib/camomile/dyn -I /home/opam/.opam/4.06/lib/camomile/lib_default -I /home/opam/.opam/4.06/lib/camomile/library -I /home/opam/.opam/4.06/lib/charInfo_width -I /home/opam/.opam/4.06/lib/core -I /home/opam/.opam/4.06/lib/core_kernel -I /home/opam/.opam/4.06/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.06/lib/fieldslib -I /home/opam/.opam/4.06/lib/jane-street-headers -I /home/opam/.opam/4.06/lib/lambda-term -I /home/opam/.opam/4.06/lib/lwt -I /home/opam/.opam/4.06/lib/lwt/unix -I /home/opam/.opam/4.06/lib/lwt_log -I /home/opam/.opam/4.06/lib/lwt_log/core -I /home/opam/.opam/4.06/lib/lwt_react -I /home/opam/.opam/4.06/lib/mmap -I /home/opam/.opam/4.06/lib/mparser -I /home/opam/.opam/4.06/lib/ocaml/threads -I /home/opam/.opam/4.06/lib/parsexp -I /home/opam/.opam/4.06/lib/pcre -I /home/opam/.opam/4.06/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.06/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.06/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.06/lib/ppx_expect/collector -I /home/opam/.opam/4.06/lib/ppx_expect/common -I /home/opam/.opam/4.06/lib/ppx_expect/config -I /home/opam/.opam/4.06/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.06/lib/ppx_inline_test/config -I /home/opam/.opam/4.06/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.06/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.06/lib/react -I /home/opam/.opam/4.06/lib/result -I /home/opam/.opam/4.06/lib/seq -I /home/opam/.opam/4.06/lib/sexplib -I /home/opam/.opam/4.06/lib/sexplib/unix -I /home/opam/.opam/4.06/lib/sexplib0 -I /home/opam/.opam/4.06/lib/spawn -I /home/opam/.opam/4.06/lib/splittable_random -I /home/opam/.opam/4.06/lib/stdio -I /home/opam/.opam/4.06/lib/typerep -I /home/opam/.opam/4.06/lib/variantslib -I /home/opam/.opam/4.06/lib/zed -no-alias-deps -open Anthill -o lib/.anthill.objs/byte/anthill__Group.cmo -c -impl lib/group.pp.ml)
# File "lib/group.ml", line 7, characters 31-45:
# Error: Unbound module Stdlib
```
This error doesn't look too bad to handle and can be resolved in two ways: use `stdlib-shims` to be compatible with OCaml 4.06 or remove compatibility with OCaml 4.06 all together.

**Comment by martindemello** [comment]
i bumped the requirement to ` "ocaml" {>= "4.07"}`; can you make the same change here or should i push a new package?

File: packages/anthill/anthill.0.1/opam [hunk 1]
```diff
@@ -3,7 +3,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "ocaml" {>= "4.06"}
+  "ocaml" {>= "4.7"}
```
Comment (martindemello) [inline packages/anthill/anthill.0.1/opam:6]: should be "4.07"

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**